### ISSUE TITLE
Error with Trix editor

### ISSUE THREAD

**Issue by mysmallidea** [ichord/At.js#470]
I'm trying to get At.js to work with Trix editor, but I'm getting an error "The given range isn't in document" when selecting an item, and the item is not inserted. 

Here is an example with runnable code on SO: http://stackoverflow.com/questions/42235418/mentions-with-trix-editor-and-at-js-error-the-given-range-isnt-in-document

Has anyone run into this? Is there an easy fix?

**Comment by mysmallidea** [comment]
The StackOverflow code runner seems to not work in Safari, but Jsfiddle works fine: https://jsfiddle.net/mysmallidea/gfaj061h/

**Comment by gertjanwytynck** [comment]
Hi @crispinheneise, same issue here. Did you find a solution?

**Comment by crispinheneise** [comment]
No, we ended up dropping At.js from this project. Sorry that's not very helpful.

**Comment by jeffAwesome** [comment]
I did create a workout to include the At.js functionality. 

    `    document.addEventListener("trix-initialize", function(event) {
        var element = $("trix-editor");
        element.on("inserted.atwho", function(event, flag, query) {
            element[0].editor.insertString(flag[0].innerHTML);
        });
    });`

**Comment by rmaspero** [comment]
I've iterated on @jeffAwesome approach as when you typed @ and then added a few characters it would append the strong tag around the part of the name you matched and leave what you had typed in the input.

Again this is pretty dirty and could defo do with some improvement, but I found it to work.

```
$(element.target).on("inserted.atwho", function(event, flag, query) {

      // this searches the returned html for the typed string
      var typedStringMatches = flag[0].innerHTML.match(/<strong>(.*)<\/strong>/);

      // this sets the string to be replaced as an empty string
      var typedString = ''

      // if something was searched it updates the typedString variable
      if(typedStringMatches != null){
        typedString = typedStringMatches[1]
      }

      // this finds the cursors current location
      var cursorLocation = event.target.editor.getSelectedRange()[0]

      // removes the typed string between the cursors location and the length of the typed string
      event.target.editor.setSelectedRange([cursorLocation - typedString.length, cursorLocation]);
      event.target.editor.deleteInDirection("backwards")

      // adds the selected matched string
      event.target.editor.insertString(flag[0].innerText);
    });
```
